File: Java/app/src/main/java/software/rsquared/template/ui/main/MainActivity.java
package software.rsquared.template.ui.main;

import android.os.Bundle;

import software.rsquared.template.R;
import software.rsquared.template.databinding.ActivityMainBinding;
import software.rsquared.template.ui.BaseFragmentActivity;

public class MainActivity extends BaseFragmentActivity<ActivityMainBinding> {

    @Override
    protected int getLayoutRes() {
        return R.layout.activity_main;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding.get().setCallback(new Callback() {
            @Override
            public void onShowFragment() {

            }

            @Override
            public void onShowPagerSample() {

            }
        });
    }

    public interface Callback {

        void onShowFragment();

        void onShowPagerSample();

    }
}

File: Kotlin/app/src/main/java/software/rsquared/template/ui/main/MainActivity.kt
package software.rsquared.template.ui.main

import android.os.Bundle
import software.rsquared.template.R
import software.rsquared.template.databinding.ActivityMainBinding
import software.rsquared.template.ui.BaseVmActivity
import software.rsquared.template.utils.navigation.launchActivity

class MainActivity : BaseVmActivity<MainViewModel, ActivityMainBinding>() {

    override fun getLayoutRes(): Int {
        return R.layout.activity_main
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        //where binding and vm are ready
    }
}
